### ISSUE TITLE
Widget Visibility: widget set to show in category view is showing in other views

### ISSUE THREAD

**Issue by eliorivero** [Automattic/jetpack#6591]
#### Steps to reproduce the issue
1. add a widget and set its Visibility to 
<img width="271" alt="captura de pantalla 2017-03-07 a las 12 01 57" src="https://cloud.githubusercontent.com/assets/1041600/23662595/df24cda0-032e-11e7-8241-26ed4fb9f652.png">

#### What I expected
To be displayed only in category views.

#### What happened instead
It's also displayed in single views.

Note that this is a regression from 4.6 where the visibility works as expected.

**Comment by zinigor** [comment]
@elio I'd say that it's not a bug, but instead a bug fix. The logic here is that whenever you choose your widget to be shown on all category pages, you also make it appear on all posts that have a category set. It was working before like that if you chose a category, but not if you chose `all categories`. #6326 has introduced the change. The actual logic change can be seen here: https://github.com/Automattic/jetpack/commit/733de0f7e6176b8d11832cbcf19da200e805f291#diff-deea61a2ffe944e4c0f7e2bc0d534be2L607

**Comment by eliorivero** [comment]
Regarding this

>  It was working before like that if you chose a category, but not if you chose all categories.

It was working as expected before with those same settings. As I pointed in Slack, I tested it with `branch-4.6` and it was ok.

That change you pointed is the one messing this. If I set a widget to be shown in all category pages, I expected it to be shown in all category pages not also in single views. Think from a user perspective: I don't care about explanations about the code, just that it was doing something before and now it's doing something different and something that is completely wrong.
And I don't know what's with the argument here

> The logic here is that whenever you choose your widget to be shown on all category pages, you also make it appear on all posts that have a category set.

that's completely, absolutely, utterly illogical to me like asking for a Borzoi and getting instead a Nevskaya Maskaradnaya. There's a reason why WP has different conditional tags `is_category`, `is_taxonomy`, `is_single`. A category view is not a single view.
